staging/src/k8s.io/apiserver/pkg/audit/request.go

```diff
@@ -42,6 +42,7 @@ func NewEventFromRequest(req *http.Request, level auditinternal.Level, attribs a
 		RequestReceivedTimestamp: metav1.NewMicroTime(time.Now()),
 		Verb:       attribs.GetVerb(),
 		RequestURI: req.URL.RequestURI(),
+		UserAgent:  req.Header.Get("User-Agent"),
```

Comment:
`Maybe`  we can use this function:
https://golang.org/pkg/net/http/#Request.UserAgent

Reply:
good idea